types/lockfile-lint-api/index.d.ts

```diff
@@ -0,0 +1,73 @@
+// Type definitions for lockfile-lint-api 5.1
+// Project: https://github.com/lirantal/lockfile-lint/tree/master/packages/lockfile-lint-api
+// Definitions by: Markus Lasermann <https://github.com/snaptags>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.9
+
+export type hosts = ReadonlyArray<string>;
+
+export interface packageMetadata {
+    version: string;
+    resolved?: string;
+    dependencies?: Record<string, string>; // e.g. {'balanced-match': '^1.0.0', 'concat-map': '0.0.1'}
+}
+
+export type packages = Record<string, packageMetadata>;
+
+export interface validationOptions {
+    emptyHostname?: boolean;
+}
+
+export interface error {
+    message: string;
+    package: string;
+}
+
+export interface validationError {
+    type: 'error';
+    errors: error[];
+}
+
+export interface validationSuccess {
+    type: 'success';
+    object: packages;
+}
+
+export type validationResult = validationError | validationSuccess;
+
+export class ValidateHost {
+    constructor(packages: { packages: packages });
+    validate: (hosts: hosts, options?: validationOptions) => validationResult;
```

Comment:
Are these supposed to be properties, or should they be defined as methods?

```suggestion
    validate(hosts: hosts, options?: validationOptions): validationResult;
```

Reply:
Thanks for pointing this out, they should indeed be methods.